sql/analyzer/apply_indexes_from_outer_scope.go

```diff
@@ -295,3 +295,149 @@ func Schemas(nodes []sql.Node) sql.Schema {
 	}
 	return schema
 }
+
+// A joinColExpr  captures a GetField expression used in a comparison, as well as some additional contextual
+// information. Example, for the base expression col1 + 1 > col2 - 1:
```

Comment:
very helpful to use an example expression here and explain how it maps to the fields in the struct!